tests/unix/schedule_thread.py

```diff
@@ -0,0 +1,38 @@
+# This test ensures that the scheduler doesn't trigger any assertions
+# while dealing with concurrent access from multiple threads.
+# (Assertions are only enabled on the Unix coverage build).
+
```

Comment:
as discussed please make this more general (eg run on esp32), so probably call it `tests/thread/stress_schedule.py`